tests/test_downloaders.py

```diff
@@ -0,0 +1,66 @@
+import hashlib
+import os
+import shutil
+import tempfile
+from unittest import TestCase
+
+from fuel.downloaders.base import (download, default_downloader,
+                                   filename_from_url)
+
+iris_url = ('https://archive.ics.uci.edu/ml/machine-learning-databases/' +
+            'iris/iris.data')
+iris_hash = "6f608b71a7317216319b4d27b4d9bc84e6abd734eda7872b71a458569e2656c0"
+
+
+class DummyArgs:
+    def __init__(self, **kwargs):
+        for key, val in kwargs.items():
+            setattr(self, key, val)
+
+
+def test_filename_from_url():
+    assert filename_from_url(iris_url) == 'iris.data'
+
+
+def test_download():
+    f = tempfile.SpooledTemporaryFile()
+    download(iris_url, f)
+    f.seek(0)
+    assert hashlib.sha256(f.read()).hexdigest() == iris_hash
```

Comment:
Yeah, it's just for file download integrity, right? Spreading a virus
through fuel-download is probably not the most efficient attack vector
anyway :p
On Apr 3, 2015 5:28 PM, "David Warde-Farley" <notifications@github.com>
wrote:

> In tests/test_downloaders.py
> <https://github.com/bartvm/fuel/pull/59#discussion_r27757134>:
>
> > +
> > +class DummyArgs:
> > +    def __init__(self, **kwargs):
> > +        for key, val in kwargs.items():
> > +            setattr(self, key, val)
> > +
> > +
> > +def test_filename_from_url():
> > +    assert filename_from_url(iris_url) == 'iris.data'
> > +
> > +
> > +def test_download():
> > +    f = tempfile.SpooledTemporaryFile()
> > +    download(iris_url, f)
> > +    f.seek(0)
> > +    assert hashlib.sha256(f.read()).hexdigest() == iris_hash
>
> I'm -0.5 on MD5, given that it has devastating security problems, but I
> suppose it's not a big risk since we're not using it to verify executables.
> … <#14c812f7497e2130_>
> On Fri, Apr 3, 2015 at 5:09 PM, Bart van Merriënboer <
> notifications@github.com> wrote: In tests/test_downloaders.py <
> https://github.com/bartvm/fuel/pull/59#discussion_r27756061>: > + >
> +class DummyArgs: > + def __init__(self, **kwargs): > + for key, val in
> kwargs.items(): > + setattr(self, key, val) > + > + > +def
> test_filename_from_url(): > + assert filename_from_url(iris_url) ==
> 'iris.data' > + > + > +def test_download(): > + f =
> tempfile.SpooledTemporaryFile() > + download(iris_url, f) > + f.seek(0) > +
> assert hashlib.sha256(f.read()).hexdigest() == iris_hash Looks good to me,
> but maybe switch to MD5? On my computer that's almost 3 times faster for a
> 1GB file (1.4 s vs 3.6 s), which might pay off for very large datasets. —
> Reply to this email directly or view it on GitHub <
> https://github.com/bartvm/fuel/pull/59/files#r27756061>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bartvm/fuel/pull/59/files#r27757134>.
>